### PR TITLE
Make go flags customisable

### PR DESCRIPTION
**Pull request by hanzei**
I've run into cases where I wanted to customise the go flags and thought this might be useful in this file.

I opted to have two different variables because there are some flags just for one command.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,6 +2,8 @@ GO ?= $(shell command -v go 2> /dev/null)
 NPM ?= $(shell command -v npm 2> /dev/null)
 CURL ?= $(shell command -v curl 2> /dev/null)
 MANIFEST_FILE ?= plugin.json
+GO_TEST_FLAGS ?= -race
+GO_BUILD_FLAGS ?=
```
Comment (hanzei) [inline Makefile]: Added this here to make clear what the comment of the variable is. 1/5

File: Makefile [hunk 2]
```diff
@@ -136,7 +138,7 @@ endif
 .PHONY: test
 test: webapp/.npminstall
 ifneq ($(HAS_SERVER),)
-	$(GO) test -race -v ./server/...
```
Comment (hanzei) [inline Makefile]: I intentionally removed `-v` here. `-v` is handy with a low number of tests, but with larger plugin it just spams the log. 2/5

OT: Maybe we can add support for `make test -v`.

File: Makefile (hunk 1 above)
Comment (ali-farooq0) [reply Makefile]: sorry, to make clear which comment?

Comment (hanzei) [reply Makefile]: Dump type. Please take another look.

**Comment by ali-farooq0** [comment]
Looks like this has a conflict now :(

File: Makefile (hunk 2 above)
Comment (hanzei) [reply Makefile]: > imagining a CI pipeline as the primary client

The usually case I had in mind was a developer that is testing his/her changes before committing. But the CI case convinced me.

**Pull request merged**